src/main/scala/scripts/PrepareOsmToR5.scala

```diff
@@ -0,0 +1,236 @@
+package scripts
+
+import java.nio.file.{Files, Paths, StandardCopyOption}
+
+import beam.sim.common.GeoUtils
+import com.conveyal.osmlib.OSM
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.Coord
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.sys.process.Process
+
+object PrepareOsmToR5 extends App {
+  if (args.length == 0) {
+    println("""
+Usage: [options] -i inputFilePath -o outputFilePath
+  -f
+    filter input file by osmium-tool
+
+  -s <value>
+    simplify ways. Node will be removed possibly if distance to next one is lesser than <value> meters
+
+  --stats
+    print OSM stats for each iteration
+""")
+
+    System.exit(1)
+  }
+
+  val argList = args.toList
+
+  def nextOption(map: Map[String, String], list: List[String]): Map[String, String] = {
+    list match {
+      case Nil                   => map
+      case "-f" :: tail          => nextOption(map ++ Map("filter" -> ""), tail)
+      case "-s" :: value :: tail => nextOption(map ++ Map("simplify" -> value), tail)
+      case "-i" :: value :: tail => nextOption(map ++ Map("input" -> value), tail)
+      case "-o" :: value :: tail => nextOption(map ++ Map("output" -> value), tail)
+      case "--stats" :: tail     => nextOption(map ++ Map("stats" -> "true"), tail)
+      case option :: tail => {
+        println(s"Unknown option $option")
+        System.exit(1)
+        Map()
+      }
+    }
+  }
+
+  val options = nextOption(Map(), argList)
+  val showStats = options.getOrElse("stats", "false").toBoolean
+  var currentOperationResultFile = ""
+
+  val preparer = new PrepareOsmToR5()
+
+  if (showStats) {
+    preparer.printOsmStats(preparer.readOsm(options("input")))
+  }
+
+  if (options.contains("filter")) {
+    currentOperationResultFile = preparer.filter(options("input"), showStats)
+    println("Filtration by osmium-tool was successfully finished")
+  }
+
+  if (options.contains("simplify")) {
+    val inputFile = if (currentOperationResultFile.isEmpty) options("input") else currentOperationResultFile
```

Comment:
Should this handle a more graceful failure of "input" not being specified? And other areas where correlations are made between options - but not explicitly checked

Reply:
Also, this could be a map

```suggestion
    val inputFile = currentOperationResultFile.getOrElse(options("input"))
```